packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -256,11 +256,11 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
       selectedDate: date,
       isDatePickerShown: false,
       formattedDate: formatDate && date ? formatDate(date) : '',
+    }, () => {
+      if (onSelectDate) {
+        onSelectDate(date);
+      }
```

Comment:
I assume you are making sure not to regress the original bug (issue #1274).

Reply:
Yes, I have verified that the original issue is not regressed.